src/Core/Models/Business/SubscriptionUpdate.cs

```diff
@@ -262,3 +261,81 @@ public override List<SubscriptionItemOptions> UpgradeItemsOptions(Subscription s
             SubscriptionItem(subscription, _existingPlanStripeId);
 
 }
+
+public class SubscriptionUpdateForSecretsManager : SubscriptionUpdate
+{
+    private readonly StaticStore.Plan _plan;
+    private readonly long? _additionalSeats;
+    private readonly long? _additionalServiceAccounts;
+    protected override List<string> PlanIds => new() { _plan.StripeSeatPlanId, _plan.StripeServiceAccountPlanId };
+    public SubscriptionUpdateForSecretsManager(StaticStore.Plan plan, long? additionalSeats, long? additionalServiceAccounts)
+    {
+        _plan = plan;
+        _additionalSeats = additionalSeats;
+        _additionalServiceAccounts = additionalServiceAccounts;
+    }
+
+    public override List<SubscriptionItemOptions> RevertItemsOptions(Subscription subscription)
+    {
+        var updatedItems = new List<SubscriptionItemOptions>();
+
+        RetrieveExistingSubsctiptionItems(subscription, updatedItems);
+        RemovePreviousSecretsManagerItems(updatedItems);
+
+        return updatedItems;
+    }
+
+    public override List<SubscriptionItemOptions> UpgradeItemsOptions(Subscription subscription)
+    {
+        var updatedItems = new List<SubscriptionItemOptions>();
+
+        RetrieveExistingSubsctiptionItems(subscription, updatedItems);
```

Comment:
I had a look at this today but I still don't understand it. Usually we just send a list of the items to be adjusted. Here we send a list of all the current items + the new item. Wouldn't Stripe just ignore the current items (no change) and add the new item, which is the same as only sending the new item?

I tried deleting this method call and it still worked for me.

Reply:
Yeah, it have also tested without the method and it works too. i have removed it. An this method can help in our Ac-1535